### PR TITLE
LB-WAF association from ingress

### PR DESCRIPTION
**Pull request by aryszka**
- [x] apply WAF Web ACL parameter
- [x] detect updates
- [x] use separate LB for ingresses with and without WAF
- [x] collapse identical stacks after WAF was removed
- [x] fix global WAF if necessary
- [x] conventional abbreviation casing
- [x] tests
- [x] docs

### CODE REVIEW

File: kubernetes/adapter.go [hunk 1]
```diff
@@ -89,7 +89,7 @@ type Ingress struct {
 	SSLPolicy        string
 	IPAddressType    string
 	LoadBalancerType string
-	WafWebACLId      string
+	WAFWebACLId      string
```
Comment (mikkeloscar) [inline kubernetes/adapter.go:92]: nit: Should the name be `WAFWebACLID` (`Id -> ID`) to follow the go conventions?

File: worker.go [hunk 2]
```diff
@@ -92,7 +92,7 @@ func (l *loadBalancer) AddIngress(certificateARNs []string, ingress *kubernetes.
 		l.securityGroup != ingress.SecurityGroup ||
 		l.sslPolicy != ingress.SSLPolicy ||
 		l.loadBalancerType != ingress.LoadBalancerType ||
-		l.http2 != ingress.HTTP2 || l.wafWebACLId != ingress.WafWebACLId {
+		l.http2 != ingress.HTTP2 {
```
Comment (mikkeloscar) [inline worker.go:95]: This `if` has the function of grouping ingresses on load balancers based on all the parameters. If you don't remove the `wafWebACLId` check, then there should not be a need for the  `groupLBsByCurrentWAF()` function below as it's effectively the same thing from what I understand, just implemented in a different way.

Comment (aryszka) [reply worker.go:95]: there was a reason during an interim state for making this change. Basically changes were not properly handled, but I need to check in detail. If possible, then I'll revert this change to make it analogous to the rest of these properites.

Comment (aryszka) [reply worker.go]: done

**Comment by mikkeloscar** [comment]
:+1:

**Comment by aryszka** [comment]
:+1:

**Pull request merged**